pkg/gofr/datasource/scylladb/scylladb.go

```diff
@@ -0,0 +1,352 @@
+func (c *Client) Query(dest any, stmt string, values ...any) error {
+	return c.QueryWithCtx(context.Background(), dest, stmt, values...)
+}
+
+// Exec executes a CQL (Cassandra Query Language) statement on a ScyllaDB cluster
+// with the provided values, using the default context (context.Background()).
+func (c *Client) Exec(stmt string, values ...any) error {
+	return c.ExecWithCtx(context.Background(), stmt, values)
+}
+
+// ExecWithCtx executes a CQL statement by using the context,statement,values and returns error.
+func (c *Client) ExecWithCtx(ctx context.Context, stmt string, values ...any) error {
+	span := c.addTrace(ctx, "exec", stmt)
+	defer c.sendOperationStats(&QueryLog{Operation: "ExecWithCtx", Query: stmt, Keyspace: c.config.Keyspace}, time.Now(), "exec", span)
+
+	return c.scylla.session.Query(stmt, values...).Exec()
+}
+
+// ExecCAS performs Compare and Set operation on ScyllaDB cluster.
+func (c *Client) ExecCAS(dest any, stmt string, values ...any) (bool, error) {
+	return c.ExecCASWithCtx(context.Background(), dest, stmt, values)
+}
+
+// ExecCASWithCtx takes default context,destination,statement,values and  return bool and error.
+//
+//nolint:exhaustive // We just want to take care of slice and struct in this case.
+func (c *Client) ExecCASWithCtx(ctx context.Context, dest any, stmt string, values ...any) (bool, error) {
+	var (
+		applied bool
+		err     error
+	)
+
+	span := c.addTrace(ctx, "exec-cas", stmt)
+
+	defer c.sendOperationStats(&QueryLog{Operation: "ExecCASWithCtx", Query: stmt, Keyspace: c.config.Keyspace}, time.Now(), "exec-cas", span)
+
+	rvo := reflect.ValueOf(dest)
+	if rvo.Kind() != reflect.Ptr {
+		c.logger.Debugf("we did not get a pointer. data is not settable.")
+
+		return false, errDestinationIsNotPointer
+	}
+
+	rv := rvo.Elem()
+	q := c.scylla.session.Query(stmt, values...)
+
+	switch rv.Kind() {
+	case reflect.Struct:
+		applied, err = c.rowsToStructCAS(q, rv)
+
+	case reflect.Slice:
+		c.logger.Debugf("a slice of %v was not expected.", reflect.SliceOf(reflect.TypeOf(dest)).String())
+
+		return false, errUnexpectedSlice{target: reflect.SliceOf(reflect.TypeOf(dest)).String()}
+
+	case reflect.Map:
+		c.logger.Debugf("a map was not expected.")
+
+		return false, errUnexpectedMap
+
+	default:
+		applied = true
+	}
+
+	return applied, err
+}
+
+// QueryWithCtx takes context ,destination,statement,values and returns error.
+//
+//nolint:exhaustive // We just want to take care of slice and struct in this case
+func (c *Client) QueryWithCtx(ctx context.Context, dest any, stmt string, values ...any) error {
+	span := c.addTrace(ctx, "query", stmt)
+
+	defer c.sendOperationStats(&QueryLog{Operation: "QueryWithCtx", Query: stmt, Keyspace: c.config.Keyspace}, time.Now(), "query", span)
+
+	rvo := reflect.ValueOf(dest)
+	if rvo.Kind() != reflect.Ptr {
+		c.logger.Error("we did not get a pointer. data is not settable.")
```

Comment:
Here you are using Error, but here a debug

https://github.com/gofr-dev/gofr/pull/1360#discussion_r1906555172